src/pages/workspaces/workspace/analysis/modals/AnalysisDuplicator.test.ts

```diff
@@ -0,0 +1,282 @@
+import '@testing-library/jest-dom'
+
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { act } from 'react-dom/test-utils'
+import { h } from 'react-hyperscript-helpers'
+import { AzureStorage, AzureStorageContract } from 'src/libs/ajax/AzureStorage'
+import { GoogleStorage, GoogleStorageContract } from 'src/libs/ajax/GoogleStorage'
+import { CloudProviderType, cloudProviderTypes } from 'src/libs/workspace-utils'
+import { defaultAzureWorkspace, defaultGoogleWorkspace } from 'src/pages/workspaces/workspace/analysis/_testData/testData'
+import {
+  AbsolutePath, AnalysisFile, getDisplayName, getExtension, getFileName, useAnalysisFiles
+} from 'src/pages/workspaces/workspace/analysis/file-utils'
+import {
+  AnalysisDuplicator,
+  AnalysisDuplicatorProps
+} from 'src/pages/workspaces/workspace/analysis/modals/AnalysisDuplicator'
+import {
+  getToolFromFileExtension, ToolLabel,
+  toolLabelTypes,
+  tools
+} from 'src/pages/workspaces/workspace/analysis/tool-utils'
+import { asMockedFn } from 'src/testing/test-utils'
+
+
+jest.mock('src/components/Modal', () => {
```

Comment:
There's now a reusable mock for Modal. See #3497.

Reply:
👀

Reply:
Hmm... it doesn't seem to pair well out of the box with `.ts` files, the usage copied from that PR breaks all tests.. investigating with Miguel

Reply:
It's not a TS issue. The problem is the `jest.resetAllMocks` in `beforeEach` which removes the implementation from the shared mock.